### PR TITLE
added before query param to Armis event collector

### PR DESCRIPTION
**Pull request by YuvHayun**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [XSUP-36261](https://jira-dc.paloaltonetworks.com/browse/XSUP-36261)

## Description
A few sentences describing the overall goals of the pull request's commits.

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/Armis/Integrations/ArmisEventCollector/ArmisEventCollector.py [hunk 1]
```diff
@@ -95,7 +96,7 @@ def fetch_by_ids_in_aql_query(self, aql_query: str, order_by: str = 'time'):
         return raw_response.get('data', {}).get('results', [])
 
     def fetch_by_aql_query(self, aql_query: str, max_fetch: int, after: datetime,
-                           order_by: str = 'time', from_param: None | int = None):
+                           order_by: str = 'time', from_param: None | int = None, before: datetime = datetime.now()):
```
Comment (JudahSchwartz) [inline Packs/Armis/Integrations/ArmisEventCollector/ArmisEventCollector.py:99]: default should be None and only add to query if exists

File: Packs/Armis/Integrations/ArmisEventCollector/ArmisEventCollector.py [hunk 2]
```diff
@@ -228,13 +233,17 @@ def calculate_fetch_start_time(last_fetch_time: datetime | str | None, fetch_sta
...
     # case 2
     elif fetch_start_time:
-        return fetch_start_time
+        if fetch_start_time > before_time:
+            last_fetch_datetime = before_time - timedelta(minutes=1)
+        return fetch_start_time, before_time
     # case 3
     else:
-        return datetime.now() - timedelta(minutes=1)
+        return before_time - timedelta(minutes=1), before_time
```
Comment (JudahSchwartz) [inline Packs/Armis/Integrations/ArmisEventCollector/ArmisEventCollector.py:246]: just return the before_time. do `- timedelta(minutes=1) ` when using this instead

File: Packs/Armis/Integrations/ArmisEventCollector/ArmisEventCollector.yml [hunk 3]
```diff
@@ -54,6 +54,14 @@ configuration:
   defaultvalue: "1"
   type: 19
   required: false
+- section: Collect
+  advanced: true
+  display: Minutes to delay
+  additionalinfo: Number of minutes to delay when fetching events (to handle events creation delay).
```
Comment (JudahSchwartz) [inline Packs/Armis/Integrations/ArmisEventCollector/ArmisEventCollector.yml:60]: make sure only audit. default should be 0

File: Packs/Armis/Integrations/ArmisEventCollector/ArmisEventCollector.py [hunk 4]
```diff
@@ -33,6 +33,7 @@ def __init__(self, unique_id_key, aql_query, type, order_by, dataset_name):
 PRODUCT = 'security'
 API_V1_ENDPOINT = '/api/v1'
 DEFAULT_MAX_FETCH = 5000
+DEFAULT_FETCH_DELAY = 10
```
[the remark this answers is not in the document]
Comment (YuvHayun) [reply Packs/Armis/Integrations/ArmisEventCollector/ArmisEventCollector.py:36]: Just a placeholder in the meantime.
I'm still waiting for a response from Armis about the expected delay time.

File: Packs/Armis/Integrations/ArmisEventCollector/ArmisEventCollector.py (hunk 2 above)
Comment (YuvHayun) [reply Packs/Armis/Integrations/ArmisEventCollector/ArmisEventCollector.py:246]: I'll make something less code duplicating.

File: Packs/Armis/Integrations/ArmisEventCollector/ArmisEventCollector.yml [hunk 5]
```diff
@@ -54,6 +54,14 @@ configuration:
   defaultvalue: "1"
   type: 19
   required: false
+- section: Collect
+  advanced: true
+  display: Minutes to delay
+  additionalinfo: Number of minutes to delay when fetching events (to handle events creation delay), default is 0.
```
Comment (JudahSchwartz) [inline Packs/Armis/Integrations/ArmisEventCollector/ArmisEventCollector.yml:60]: should mention only for audit

File: Packs/Armis/Integrations/ArmisEventCollector/ArmisEventCollector.yml [hunk 6]
```diff
@@ -54,6 +54,14 @@ configuration:
...
+  name: fetch_delay
+  defaultvalue: "0"
+  type: 0
```
Comment (JudahSchwartz) [inline Packs/Armis/Integrations/ArmisEventCollector/ArmisEventCollector.yml:63]: thinking now maybe the default shouldnt be 0, @ShahafBenYakir wdyt should we have the delay be default behavior?

File: Packs/Armis/ReleaseNotes/1_1_15.md [hunk 7]
```diff
@@ -4,4 +4,4 @@
 
 ##### Armis Event Collector
 
-- Added the **Minutes to delay** parameter to handle an issue where events were missing due to delays in Armis API.
+- Added the **Minutes to delay** parameter to handle an issue where events were missing due to delays in Armis API. Note that the default value is 10 minutes but a higher value might be needed for users with heavier traffic.
```
Comment (JudahSchwartz) [inline Packs/Armis/ReleaseNotes/1_1_15.md:7]:
```suggestion
- Added the **Minutes to delay** parameter to handle an issue where events were missing due to delays in Armis API. The default value is 10 minutes. A higher value can be configured in the event that some events are still not available within those ten minutes. 
```

File: Packs/Armis/Integrations/ArmisEventCollector/ArmisEventCollector.yml [hunk 8]
```diff
@@ -57,9 +57,9 @@ configuration:
 - section: Collect
   advanced: true
   display: Minutes to delay
-  additionalinfo: Number of minutes to delay when fetching events (to handle events creation delay), default is 0.
+  additionalinfo: Number of minutes to delay when fetching events (to handle events creation delay in the Armis database). Default is 10 minutes but note a higher value might be needed for users with heavier traffic.
```
Comment (JudahSchwartz) [inline Packs/Armis/Integrations/ArmisEventCollector/ArmisEventCollector.yml:60]: same change here

**Review by JudahSchwartz — APPROVED** [review]

**Pull request merged**